### PR TITLE
NISP-2297: Purge NI record

### PR DESCRIPTION
**Pull request by prashantchoudhari**

### CODE REVIEW

**Comment by prashantchoudhari** [comment]
Hello @dspork - Please review the last commit.

File: app/uk/gov/hmrc/nationalinsurancerecord/connectors/NpsConnector.scala [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+package uk.gov.hmrc.nationalinsurancerecord.connectors
+
+import uk.gov.hmrc.nationalinsurancerecord.domain.{NationalInsuranceRecord, NationalInsuranceTaxYear}
+
+import scala.concurrent.Future
+
+trait NpsConnector {
+
+  def getNationalInsuranceRecord: Future[NationalInsuranceRecord]
```
Comment (dspork) [inline app/uk/gov/hmrc/nationalinsurancerecord/connectors/NpsConnector.scala]: This should return the NPS Model, not the formatted model.

File: app/uk/gov/hmrc/nationalinsurancerecord/domain/nps/NpsLiability.scala [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+package uk.gov.hmrc.nationalinsurancerecord.domain.nps
+
+import org.joda.time.LocalDate
+import play.api.libs.functional.syntax._
+import play.api.libs.json._
+
+case class NpsLiability(
+                         liabilityType: Int,
+                         startYear: Option[LocalDate],
```
Comment (dspork) [inline app/uk/gov/hmrc/nationalinsurancerecord/domain/nps/NpsLiability.scala]: Are these dates required?

File: app/uk/gov/hmrc/nationalinsurancerecord/util/NIRecordConstants.scala [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.nationalinsurancerecord.util
+
+class NIRecordConstants {
```
Comment (dspork) [inline app/uk/gov/hmrc/nationalinsurancerecord/util/NIRecordConstants.scala]: We should remove this empty class

File: test/uk/gov/hmrc/nationalinsurancerecord/services/NationalInsuranceRecordServiceSpec.scala [hunk 4]
```diff
@@ -193,4 +197,161 @@ class NationalInsuranceRecordServiceSpec extends NationalInsuranceRecordUnitSpec
...
+        }
+      }
+      "return classThreePayableByPenalty to be None for taxyear 2015-16"  in {
+        whenReady(niRecordF) { ni =>
+          ni.taxYears.head.classThreePayableByPenalty shouldBe None
+          ni.taxYears(2).classThreePayableByPenalty shouldBe Some(new LocalDate(2023,4,5))
+        }
+      }
+
+      "return payable and underinvestigateion flag to be false for taxyear 2015-16"  in {
```
Comment (dspork) [inline test/uk/gov/hmrc/nationalinsurancerecord/services/NationalInsuranceRecordServiceSpec.scala]: underinvestigateion should be underInvestigation

File: app/uk/gov/hmrc/nationalinsurancerecord/connectors/NpsConnector.scala [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+  def getTaxYear: Future[NationalInsuranceTaxYear]
```
Comment (dspork) [inline app/uk/gov/hmrc/nationalinsurancerecord/connectors/NpsConnector.scala]: Should this method exist? NPS does not have a single tax year endppoint.
NationalInsuranceService should pull a NationalInsuranceTaxYear out from the NpsNIRecord that comes back.

File: app/uk/gov/hmrc/nationalinsurancerecord/util/NIRecordConstants.scala (hunk 3 above)
Comment (prashantchoudhari) [inline app/uk/gov/hmrc/nationalinsurancerecord/util/NIRecordConstants.scala]: This is deleted already, but will put in next commit.

File: app/uk/gov/hmrc/nationalinsurancerecord/domain/nps/NpsLiability.scala (hunk 2 above)
Comment (prashantchoudhari) [inline app/uk/gov/hmrc/nationalinsurancerecord/domain/nps/NpsLiability.scala]: These are deleted, but will put in next commit.

File: app/uk/gov/hmrc/nationalinsurancerecord/connectors/NpsConnector.scala (hunk 5 above)
Comment (prashantchoudhari) [inline app/uk/gov/hmrc/nationalinsurancerecord/connectors/NpsConnector.scala]: Agreed, we dont need another method in trait.

File: app/uk/gov/hmrc/nationalinsurancerecord/services/NationalInsuranceRecordService.scala [hunk 6]
```diff
@@ -104,10 +111,122 @@ trait NispConnection extends NationalInsuranceRecordService {
...
+      } else {
+        purgedNIRecord.niTaxYears.map(npsTaxYearToNIRecordTaxYear).find(x => x.taxYear == taxYear.taxYear) match {
+          case Some(nationalInsuranceRecordTaxYear) => Right(nationalInsuranceRecordTaxYear)
+          case _ => throw new NotFoundException(s"taxYear ${taxYear.taxYear} Not Found for $nino")
+        }
+      }
+    }
+  }
+
+  def homeResponsibilitiesProtection(liabilities: List[NpsLiability]): Boolean =
```
Comment (dspork) [inline app/uk/gov/hmrc/nationalinsurancerecord/services/NationalInsuranceRecordService.scala]: This method needs unit tests.

File: app/uk/gov/hmrc/nationalinsurancerecord/services/NationalInsuranceRecordService.scala [hunk 7]
```diff
@@ -104,10 +111,122 @@ trait NispConnection extends NationalInsuranceRecordService {
...
+    liabilities.exists(liability => NIRecordConstants.homeResponsibilitiesProtectionTypes.contains(liability.liabilityType))
+
+  def calcPre75QualifyingYears(pre75Contributions: Int, dateOfEntry: LocalDate, dateOfBirth: LocalDate): Option[Int] = {
```
Comment (dspork) [inline app/uk/gov/hmrc/nationalinsurancerecord/services/NationalInsuranceRecordService.scala]: This method needs unit tests.

File: test/uk/gov/hmrc/nationalinsurancerecord/services/NationalInsuranceRecordServiceSpec.scala [hunk 8]
```diff
@@ -193,4 +247,201 @@ class NationalInsuranceRecordServiceSpec extends NationalInsuranceRecordUnitSpec
...
+        whenReady(niRecordF) { ni =>
+          ni.numberOfGapsPayable shouldBe 1
+        }
+      }
+      "return date of entry to be 1969/8/1"  in {
+        whenReady(niRecordF) { ni =>
+          ni.dateOfEntry shouldBe new LocalDate(1969,8,1)
+        }
+      }
+      "return homeResponsibilities to be false"  in {
```
Comment (dspork) [inline test/uk/gov/hmrc/nationalinsurancerecord/services/NationalInsuranceRecordServiceSpec.scala]: This should be "to be true"

File: test/uk/gov/hmrc/nationalinsurancerecord/domain/nps/NpsSummarySpec.scala [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+
+class NpsSummarySpec extends UnitSpec {
+  // scalastyle:off magic.number
+
+  val summary = Json.parse(
+    """
+  |{
+    |    "contracted_out_flag": 1,
+    | "sensitive_flag": 0,
+    | "spa_date": "2017-11-21",
```
Comment (dspork) [inline test/uk/gov/hmrc/nationalinsurancerecord/domain/nps/NpsSummarySpec.scala]: Could you format this file, including the code and json string?

**Pull request merged**